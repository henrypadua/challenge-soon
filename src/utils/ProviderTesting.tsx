import { NativeBaseProvider, NativeBaseProviderProps } from 'native-base';

import { theme } from '@styles/theme';

export const ProviderTesting = (props: NativeBaseProviderProps) => {
	const inset = {
		frame: { x: 0, y: 0, width: 0, height: 0 },
		insets: { top: 0, left: 0, right: 0, bottom: 0 },
	};

	return (
		<NativeBaseProvider initialWindowMetrics={inset} theme={theme}>
			{props.children}
		</NativeBaseProvider>
	);
};
