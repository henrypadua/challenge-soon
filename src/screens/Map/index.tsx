import React, { useState, useRef } from 'react';
import { Dimensions } from 'react-native';

import { GOOGLE_MAPS_APIKEY } from '@env';
import { useMap } from '@hooks/useMap';
import { Button, Image, Spinner, Text, VStack } from 'native-base';
import MapView, { Marker } from 'react-native-maps';
import MapViewDirections from 'react-native-maps-directions';

import utils from '@utils/Utils';

import { ModalInfo } from '@components/ModalInfo';

import { mapStyle } from '@styles/mapStyle';

export function Map() {
	const [isReady, setIsReady] = useState(false);
	const [angle, setAngle] = useState(0);
	const [infoRoute, setInfoRoute] = useState({
		distance: 0,
		duration: 0,
	});
	const [openModal, setOpenModal] = useState(false);

	const mapView = useRef<MapView>(null);
	const { isLoading, origin, destination } = useMap();

	return (
		<>
			{isLoading ? (
				<VStack
					flex={1}
					justifyContent="center"
					alignItems="center"
					bgColor={'#242E42'}
				>
					<Spinner size={'lg'} color="#ffffff" />
					<Text color="#ffffff">Loading...</Text>
				</VStack>
			) : (
				<MapView
					ref={mapView}
					style={{
						flex: 1,
						width: Dimensions.get('window').width,
						height: Dimensions.get('window').height,
					}}
					customMapStyle={mapStyle}
					region={{
						latitude: origin?.latitude,
						longitude: origin?.longitude,
						latitudeDelta: 0.0922,
						longitudeDelta: 0.0421,
					}}
					loadingBackgroundColor="#242E42"
					loadingIndicatorColor="#ffffff"
					loadingEnabled
					onMapLoaded={() => {
						setIsReady(true);
						setOpenModal(true);
					}}
					showsBuildings={false}
					showsPointsOfInterest={false}
				>
					{origin && (
						<Marker
							coordinate={origin}
							tracksViewChanges={true}
							anchor={{
								x: 0.5,
								y: 0.5,
							}}
						>
							<Image
								source={require('@assets/pin.png')}
								height={'50px'}
								width={'32px'}
								alt="pin"
							/>
						</Marker>
					)}

					{destination && (
						<Marker
							coordinate={destination}
							tracksViewChanges={true}
							anchor={{
								x: 0.5,
								y: 0.5,
							}}
							rotation={angle}
						>
							<Image
								source={require('@assets/car.png')}
								height={'50px'}
								width={'32px'}
								alt="pin"
							/>
						</Marker>
					)}
					<MapViewDirections
						origin={origin}
						destination={destination}
						apikey={GOOGLE_MAPS_APIKEY}
						strokeWidth={5}
						strokeColor="#9722FB"
						optimizeWaypoints={true}
						onReady={(result) => {
							setInfoRoute({
								distance: result.distance,
								duration: result.duration,
							});

							if (!isReady) {
								// Fit the map based on the route
								mapView?.current?.fitToCoordinates(result.coordinates, {
									edgePadding: {
										right: Dimensions.get('window').width * 0.1,
										bottom: 400,
										left: Dimensions.get('window').width * 0.1,
										top: Dimensions.get('window').height * 0.1,
									},
								});
								// Reposition the navigator
								if (result.coordinates.length >= 2) {
									const calculatedAngle = utils.calculateAngle(
										result.coordinates,
									);
									setAngle(calculatedAngle);
								}
								setIsReady(true);
							}
						}}
					/>
				</MapView>
			)}

			<ModalInfo
				infoRoute={infoRoute}
				openModal={openModal}
				setOpenModal={setOpenModal}
			/>
		</>
	);
}
