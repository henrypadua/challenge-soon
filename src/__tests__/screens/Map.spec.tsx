import { render } from '@testing-library/react-native';

import { ProviderTesting } from '@utils/ProviderTesting';

import { Map } from '@screens/Map';

jest.useFakeTimers();

test('renders correctly', () => {
	const screen = render(
		<ProviderTesting>
			<Map />
		</ProviderTesting>,
	).toJSON();

	expect(screen).toMatchSnapshot();
});
