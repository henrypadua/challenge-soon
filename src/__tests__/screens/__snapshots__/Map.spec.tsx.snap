// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <View />,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "display": "none",
        },
      ]
    }
  >
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateY": 0,
            },
            {
              "translateX": 0,
            },
            {
              "scale": 1,
            },
            {
              "scaleX": 1,
            },
            {
              "scaleY": 1,
            },
            {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View
        accessible={false}
        collapsable={false}
        dataSet={{}}
        focusable={true}
        importantForAccessibility="no"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "rgb(0, 0, 0)",
            "bottom": 0,
            "left": 0,
            "opacity": 0.1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateY": 0,
            },
            {
              "translateX": 0,
            },
            {
              "scale": 1,
            },
            {
              "scaleX": 1,
            },
            {
              "scaleY": 1,
            },
            {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View
        closeOnOverlayClick={true}
        dataSet={{}}
        pointerEvents="box-none"
        style={
          {
            "alignItems": "center",
            "bottom": 16,
            "height": "100%",
            "justifyContent": "flex-end",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={{}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 9999,
                "elevation": 3,
                "flexDirection": "row",
                "justifyContent": "center",
                "left": 16,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "position": "absolute",
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.22,
                "shadowRadius": 2.22,
                "top": 128,
                "zIndex": 20,
              },
              {
                "height": 44,
                "width": 44,
              },
            ]
          }
        >
          <View
            dataSet={{}}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <Image
              accessibilityLabel="close"
              alt="close"
              dataSet={{}}
              onError={[Function]}
              source={1}
              style={
                [
                  {
                    "color": "#fafafa",
                    "height": 40,
                    "maxWidth": "100%",
                    "width": 40,
                  },
                  {
                    "height": 30,
                    "width": 30,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          accessibilityViewIsModal={true}
          aria-modal="true"
          dataSet={{}}
          onAccessibilityEscape={[Function]}
          style={
            {
              "backgroundColor": "#fafafa",
              "borderRadius": 8,
              "elevation": 2,
              "maxHeight": 1184,
              "maxWidth": 580,
              "overflow": "hidden",
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 1.41,
              "width": "90%",
            }
          }
        >
          <View
            dataSet={{}}
            style={
              {
                "backgroundColor": "#F7F7F7",
                "borderBottomWidth": 1,
                "borderColor": "#d4d4d4",
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#a1a1aa",
                    "borderColor": "#27272a",
                    "borderRadius": 9999,
                    "height": 48,
                    "justifyContent": "center",
                    "position": "relative",
                    "width": 50,
                  }
                }
              >
                <Image
                  accessibilityLabel="--"
                  alt="--"
                  dataSet={{}}
                  onError={[Function]}
                  source={1}
                  style={
                    [
                      {
                        "borderRadius": 9999,
                        "maxWidth": "100%",
                      },
                      {
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                />
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "flex-start",
                  }
                }
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#242E42",
                      "fontFamily": "Inter_500Medium",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Gregory Smith
                </Text>
                <View
                  dataSet={{}}
                  style={{}}
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignContent": "center",
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text />
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#C8C7CC",
                        "fontFamily": "Inter_400Regular",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    4.9
                  </Text>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                dataSet={{}}
                style={
                  {
                    "alignContent": "center",
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessible={true}
                  collapsable={false}
                  colorScheme="primary"
                  dataSet={{}}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#4252FF",
                      "borderRadius": 9999,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginLeft": "auto",
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                      "paddingTop": 10,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 16,
                    }
                  }
                />
                <View
                  accessibilityRole="button"
                  accessible={true}
                  collapsable={false}
                  colorScheme="primary"
                  dataSet={{}}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#4CE5B1",
                      "borderRadius": 9999,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginLeft": "auto",
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                      "paddingTop": 10,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
          <RCTScrollView
            contentContainerStyle={
              [
                {},
                {
                  "dataSet": {},
                },
                {},
              ]
            }
            dataSet={{}}
            style={{}}
          >
            <View>
              <View
                dataSet={{}}
                style={
                  {
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "flex": 1,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={
                      {
                        "alignContent": "center",
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Image
                      accessibilityLabel="circle"
                      alt="circle"
                      dataSet={{}}
                      onError={[Function]}
                      source={1}
                      style={
                        {
                          "maxWidth": "100%",
                        }
                      }
                    />
                    <View
                      dataSet={{}}
                      style={{}}
                    />
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#242E42",
                          "fontFamily": "Inter_500Medium",
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "marginLeft": 16,
                          "textDecorationLine": undefined,
                        }
                      }
                    />
                  </View>
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <Image
                    accessibilityLabel="line"
                    alt="line"
                    dataSet={{}}
                    onError={[Function]}
                    source={1}
                    style={
                      {
                        "left": 8,
                        "maxWidth": "100%",
                      }
                    }
                  />
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <View
                    dataSet={{}}
                    style={
                      {
                        "alignContent": "center",
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Image
                      accessibilityLabel="square"
                      alt="square"
                      dataSet={{}}
                      onError={[Function]}
                      source={1}
                      style={
                        {
                          "maxWidth": "100%",
                        }
                      }
                    />
                    <View
                      dataSet={{}}
                      style={{}}
                    />
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#242E42",
                          "fontFamily": "Inter_500Medium",
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "marginLeft": 16,
                          "textDecorationLine": undefined,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
          <View
            dataSet={{}}
            style={
              {
                "backgroundColor": "#fafafa",
                "borderColor": "#d4d4d4",
                "borderTopWidth": 1,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-end",
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingBottom": 24,
                    "paddingTop": 24,
                  }
                }
              >
                <Image
                  accessibilityLabel="car"
                  alt="car"
                  dataSet={{}}
                  onError={[Function]}
                  source={1}
                  style={
                    {
                      "maxWidth": "100%",
                    }
                  }
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 20,
                    }
                  }
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                    }
                  }
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#C8C7CC",
                        "fontFamily": "Inter_400Regular",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    DISTANCE
                  </Text>
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#242E42",
                        "fontFamily": "Inter_500Medium",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 24,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    0.0
                    km
                  </Text>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 20,
                    }
                  }
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                    }
                  }
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#C8C7CC",
                        "fontFamily": "Inter_500Medium",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    TIME
                  </Text>
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#242E42",
                        "fontFamily": "Inter_500Medium",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 24,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    00:00:00
                  </Text>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 20,
                    }
                  }
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                    }
                  }
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#C8C7CC",
                        "fontFamily": "Inter_400Regular",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    PRICE
                  </Text>
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#242E42",
                        "fontFamily": "Inter_500Medium",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 24,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    $
                    0.00
                  </Text>
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#242E42",
                    "borderRadius": 8,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#fff",
                          "fontFamily": "Inter_500Medium",
                          "fontSize": 16,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 24,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Cancel Request
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
